### ISSUE TITLE
Is it normal for the orth loss to keep decreasing？

### ISSUE THREAD

**Issue by YuKDseele** [qingshi9974/AuxT#1]
Awesome work! Thank you for providing the open-source code.
However, I encountered some problem when applying it to my own model and dataset, and I would like to know if this is normal.
I found that the convergence speed of RD Loss did not meet expectations. 
After 2M iterations training (without learning rate decay), the loss of my model is 0.967. 
When training with AuxT for 1M iterations (without learning rate decay), at 0.68M iterations, the loss is 0.979, and overall, it seems to stabilize. 

There does not appear to be any trend of the loss decreasing to 0.967 or below. What is even more concerning is that the orthloss has been decaying slowly. 
Is it normal for the orth loss to keep decreasing?

Below are the logs.
Test epoch 0: Average losses:	Loss: 9.598 |	MSE loss: 0.003 |	Bpp loss: 1.09 |	orth loss: 0.3141 |	Aux loss: 7795.75
Test epoch 100: Average losses:	Loss: 1.261 |	MSE loss: 0.000 |	Bpp loss: 0.72 |	orth loss: 0.0621 |	Aux loss: 5.81
Test epoch 200: Average losses:	Loss: 1.108 |	MSE loss: 0.000 |	Bpp loss: 0.67 |	orth loss: 0.0322 |	Aux loss: 6.46
Test epoch 300: Average losses:	Loss: 1.099 |	MSE loss: 0.000 |	Bpp loss: 0.65 |	orth loss: 0.0221 |	Aux loss: 8.39
Test epoch 400: Average losses:	Loss: 1.035 |	MSE loss: 0.000 |	Bpp loss: 0.62 |	orth loss: 0.0178 |	Aux loss: 9.82
Test epoch 500: Average losses:	Loss: 1.028 |	MSE loss: 0.000 |	Bpp loss: 0.61 |	orth loss: 0.0153 |	Aux loss: 10.07
Test epoch 600: Average losses:	Loss: 1.020 |	MSE loss: 0.000 |	Bpp loss: 0.60 |	orth loss: 0.0139 |	Aux loss: 11.68
Test epoch 700: Average losses:	Loss: 1.011 |	MSE loss: 0.000 |	Bpp loss: 0.60 |	orth loss: 0.0129 |	Aux loss: 14.43
Test epoch 800: Average losses:	Loss: 1.005 |	MSE loss: 0.000 |	Bpp loss: 0.60 |	orth loss: 0.0122 |	Aux loss: 13.49
Test epoch 900: Average losses:	Loss: 0.999 |	MSE loss: 0.000 |	Bpp loss: 0.59 |	orth loss: 0.0117 |	Aux loss: 13.40
Test epoch 1000: Average losses:	Loss: 0.994 |	MSE loss: 0.000 |	Bpp loss: 0.59 |	orth loss: 0.0112 |	Aux loss: 12.78
Test epoch 1100: Average losses:	Loss: 0.993 |	MSE loss: 0.000 |	Bpp loss: 0.59 |	orth loss: 0.0107 |	Aux loss: 14.45
Test epoch 1200: Average losses:	Loss: 0.990 |	MSE loss: 0.000 |	Bpp loss: 0.59 |	orth loss: 0.0102 |	Aux loss: 13.62
Test epoch 1300: Average losses:	Loss: 0.990 |	MSE loss: 0.000 |	Bpp loss: 0.59 |	orth loss: 0.0098 |	Aux loss: 13.73
Test epoch 1400: Average losses:	Loss: 0.990 |	MSE loss: 0.000 |	Bpp loss: 0.59 |	orth loss: 0.0094 |	Aux loss: 15.70
Test epoch 1500: Average losses:	Loss: 0.987 |	MSE loss: 0.000 |	Bpp loss: 0.58 |	orth loss: 0.0091 |	Aux loss: 14.59
Test epoch 1600: Average losses:	Loss: 0.982 |	MSE loss: 0.000 |	Bpp loss: 0.58 |	orth loss: 0.0088 |	Aux loss: 13.26
Test epoch 1700: Average losses:	Loss: 0.985 |	MSE loss: 0.000 |	Bpp loss: 0.58 |	orth loss: 0.0086 |	Aux loss: 14.42
Test epoch 1800: Average losses:	Loss: 0.980 |	MSE loss: 0.000 |	Bpp loss: 0.58 |	orth loss: 0.0084 |	Aux loss: 13.34
Test epoch 1900: Average losses:	Loss: 0.977 |	MSE loss: 0.000 |	Bpp loss: 0.58 |	orth loss: 0.0082 |	Aux loss: 13.70
Test epoch 2000: Average losses:	Loss: 0.979 |	MSE loss: 0.000 |	Bpp loss: 0.58 |	orth loss: 0.0081 |	Aux loss: 12.17
Test epoch 2100: Average losses:	Loss: 0.979 |	MSE loss: 0.000 |	Bpp loss: 0.58 |	orth loss: 0.0079 |	Aux loss: 14.60
Test epoch 2200: Average losses:	Loss: 0.975 |	MSE loss: 0.000 |	Bpp loss: 0.58 |	orth loss: 0.0078 |	Aux loss: 12.63
Test epoch 2300: Average losses:	Loss: 0.976 |	MSE loss: 0.000 |	Bpp loss: 0.58 |	orth loss: 0.0076 |	Aux loss: 13.96
Test epoch 2400: Average losses:	Loss: 0.975 |	MSE loss: 0.000 |	Bpp loss: 0.57 |	orth loss: 0.0074 |	Aux loss: 12.66
Test epoch 2500: Average losses:	Loss: 0.979 |	MSE loss: 0.000 |	Bpp loss: 0.58 |	orth loss: 0.0073 |	Aux loss: 15.12

![Image](https://github.com/user-attachments/assets/14d8110a-31b6-4ae2-b0ba-af2ec9c2d9d5)

![Image](https://github.com/user-attachments/assets/24cafd9c-468c-43b5-a017-82f9208dec57)

**Comment by YuKDseele** [comment]
After calculating the orthloss from the provided log file, I found that the trend does not match the actual trend of the orthloss during my training. So, in reality, the orthloss should stabilize rather than continuously decrease?
As shown in the image below, the orthloss calculated from the logs decreased by less than 5% between 0.2M to 0.9M steps. However, in my test, the orthloss decreased by 40% between 0.2M and 0.68M steps! Does this mean that the coefficient for orthloss should be appropriately reduced?
![Image](https://github.com/user-attachments/assets/be4935ba-2256-4df6-93af-0da8450e1778)

**Comment by qingshi9974** [comment]
I conducted experiments on TCM-small where the Anchor model achieved a test RD loss of 1.084 at 1.75M iterations before learning rate decay, while the AuxT model achieved the same RD loss of 1.084 at 0.61M iterations. Have you plotted a comparison of the corresponding RD loss convergence curves? Alternatively, you could try applying learning rate decay to see if the results meet expectations.
Regarding the ortho loss, I believe it is normal for it to decrease continuously during training as it is related to the network parameters and not the test set. I have tested ortho_weight = 0.1 on TCM, mbt2018mean, and stf and found it to be consistently effective. You can try reducing the weight in your model to see if it further improves overall performance.

![Image](https://github.com/user-attachments/assets/8fb42207-071f-4727-a347-9c04ba2a3184)
